src/os/bluestore/BlueFS.h

```diff
@@ -560,9 +567,18 @@ class BlueFS {
     int r = _flush(h, force, l);
     ceph_assert(r == 0);
   }
-  void try_flush(FileWriter *h) {
-    if (h->get_buffer_length() >= cct->_conf->bluefs_min_flush_size) {
-      flush(h, true);
+
+  void append_try_flush(FileWriter *h, const char* buf, size_t len) {
+    while (len > 0) {
```

Comment:
I think that this loop is good, but using
`std::numeric_limits<unsigned>::max()` instead of UINT_MAX is better.